brats/transformations.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+CHANNELS = 3
+
+
+class HistogramMatchingTransform:
+    """
+    Transformation performing histogram matching of volumes
+    """
+    def __init__(self, template: np.ndarray):
+        """
+
+        Args:
+            template (np.ndarray): Template which will be used for histogram
+                matching in future calls.
+                Dimensions - (Height, Width, Depth, Channel)
+        """
+        self.template = template
+
+    def __call__(self, volume: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            volume (np.ndarray): Volume to be matched.
+                Dimensions - (Height, Width, Depth, Channel)
+
+        Returns:
+             np.ndarray: Normalized volume with the same shape as input volume
+        """
+        assert self.template.ndim == volume.ndim
+        result = np.zeros_like(volume)
+        for channel in range(volume.shape[CHANNELS]):
+            result[..., channel] = self.hist_match(volume[..., channel],
+                                                   self.template[..., channel])
+        return result
+
+    @staticmethod
+    def hist_match(source: np.ndarray, template: np.ndarray) -> np.ndarray:
+        """
+            Adjust the pixel values of an image such that its histogram
+            matches that of a target image
+
+            Args:
+                source (np.ndarray): Image to transform; the histogram is
+                    computed over the flattened array
+                template (np.ndarray): Template image; can have different
+                    dimensions to source
+            Returns:
+                np.ndarray: The transformed output image
+                    of the same shape as input
+            """
+        nonzero = source > 0
+        result = np.zeros_like(source)
+        source = source[source > 0].ravel()
```

Comment:
Can image have negative values? Vould you explain this part?

Reply:
Oh okay so its non-zero and since image cant have negative values u just call bigger_than `>` operator. I think cleare would be `positive` name for `non-zero` variable.

Reply:
Changed